### DISCUSSION TITLE
Duplicate UX

### DISCUSSION THREAD

**Discussion by wherewego-nina** [payloadcms/payload#593] [Feature Requests & Ideas]
When you duplicate a document, it very quickly takes you to the create page with all the filled information that looks exactly like the edit page you're coming from, making it hard to realize that an action has occurred. It would be cool if there was some visual cue that you are now on a duplicate. Maybe adding "Copy of xxxx" to the field being used as the title?? Maybe some sort of flag at the beginning of the page? a pop up confirmation before moving on? Not sure what the best solution is, but something small to help announce this action would be great.

**Comment by jmikrut** [comment]
Hey @wherewego-nina — one of TRBL's clients actually is asking for something similar too. I think one way to do this is to build a function that allows you to _**modify**_ data before a document is duplicated, however you want.

We'll certainly build this. It should be fairly trivial. With this type of function, you could modify a document however you want before you duplicate.

What do you think?

**Comment by wherewego-nina** [comment]
I think that’s perfect!

On Tue, May 24, 2022 at 10:43 PM James Mikrut ***@***.***>
wrote:

> Hey @wherewego-nina <https://github.com/wherewego-nina> — one of TRBL's
> clients actually is asking for something similar too. I think one way to do
> this is to build a function that allows you to *modify* data before a
> document is duplicated, however you want.
>
> We'll certainly build this. It should be fairly trivial. With this type of
> function, you could modify a document however you want before you duplicate.
>
> What do you think?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/payloadcms/payload/discussions/593#discussioncomment-2816466>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY3HMM5NK25AQAR3INA6BRDVLWHTXANCNFSM5W3I4ALQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>
-- 

*     Nina Horne - Director of Web Development*
     wherewego.org

**Comment by zubricks** [comment]
Hey for everyone following here, this is now complete. You can use the `beforeDuplicate` admin hook to modify the shape of a document's data before it is duplicated. Docs here: https://payloadcms.com/docs/configuration/collections#admin-hooks

**Comment by wherewego-nina** [comment]
This is wonderful! Thank you
